pkg/utils/k8sutil.go

```diff
@@ -1088,6 +1088,15 @@ func EnsureFuncDeployment(client kubernetes.Interface, funcObj *kubelessApi.Func
 	}
 	dpm.Spec.Template.Spec.Containers[0].LivenessProbe = livenessProbe
 
+	// Add security context
+	runtimeUser := int64(1000)
+	if dpm.Spec.Template.Spec.SecurityContext == nil {
+		dpm.Spec.Template.Spec.SecurityContext = &v1.PodSecurityContext{
+			RunAsUser: &runtimeUser,
```

Comment:
Why not just set `runAsNonRoot` to true?

Reply:
we need to set the `fsGroup` for setting the runtime user as owner of the volumes we create (so it can install dependencies and write in the `/kubeless` volume)

Reply:
Also `runAsNonRoot` just prevent the pod from starting if the user is `root` it doesn't actually change the running user

Reply:
Got it. Make sense.